blocks/table/table.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Table Block
+ * Recreate a table
+ * https://www.hlx.live/developer/block-collection/table
+ */
+
+function buildCell(rowIndex) {
+  const cell = rowIndex ? document.createElement('td') : document.createElement('th');
+  if (!rowIndex) cell.setAttribute('scope', 'col');
+  return cell;
+}
+
+export default async function decorate(block) {
+  const table = document.createElement('table');
+  const thead = document.createElement('thead');
+  const tbody = document.createElement('tbody');
+  table.append(thead, tbody);
```

Comment:
instead of creating a table element you can use the columns block from milo library

Example: https://main--milo--adobecom.hlx.page/docs/library/blocks/columns
Code: https://github.com/adobecom/milo/blob/main/libs/blocks/columns/